ionoscloud/resource_dbaas_mariadb_cluster.go

```diff
@@ -0,0 +1,237 @@
+						},
+						"password": {
+							Type:             schema.TypeString,
+							Description:      "The password for a MariaDB user.",
+							Required:         true,
+							Sensitive:        true,
+							ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+						},
+					},
+				},
+			},
+			"maintenance_window": {
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Description: "A weekly 4 hour-long window, during which maintenance might occur.",
+				Optional:    true,
+				Computed:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"time": {
+							Type:             schema.TypeString,
+							Description:      "Start of the maintenance window in UTC time.",
+							Required:         true,
+							ValidateDiagFunc: validation.ToDiagFunc(validation.StringIsNotWhiteSpace),
+						},
+						"day_of_the_week": {
+							Type:             schema.TypeString,
+							Description:      "The name of the week day.",
+							Required:         true,
+							ValidateDiagFunc: validation.ToDiagFunc(validation.IsDayOfTheWeek(true)),
+						},
+					},
+				},
+			},
+			"dns_name": {
+				Type:        schema.TypeString,
+				Description: "The DNS name pointing to your cluster.",
+				Computed:    true,
+			},
+		},
+		Timeouts: &resourceDefaultTimeouts,
+	}
+}
+
+func mariaDBClusterCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).MariaDBClient
+
+	cluster, err := mariadb.GetMariaDBClusterDataCreate(d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	response, _, err := client.CreateCluster(ctx, *cluster)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("an error occured while creating a DBaaS MariaDB cluster: %w", err))
+	}
+	clusterID := *response.Id
+	d.SetId(clusterID)
+
+	err = utils.WaitForResourceToBeReady(ctx, d, client.IsClusterReady)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("error occured while checking the status for MariaDB cluster with ID: %v, error: %w", clusterID, err))
+	}
+	if err := client.SetMariaDBClusterData(d, response); err != nil {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func mariaDBClusterDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(services.SdkBundle).MariaDBClient
+	clusterID := d.Id()
+	_, apiResponse, err := client.DeleteCluster(ctx, d.Id())
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			d.SetId("")
+			return nil
+		}
+		return diag.FromErr(fmt.Errorf("error while deleting MariaDB cluster with ID: %v, error: %w", clusterID, err))
+	}
+	err = utils.WaitForResourceToBeDeleted(ctx, d, client.IsClusterDeleted)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("deletion check failed for MariaDB cluster with ID: %v, error: %w", clusterID, err))
+	}
+
+	// wait 15 seconds after the deletion of the cluster, for the lan to be freed
+	time.Sleep(constant.SleepInterval * 10)
```

Comment:
we should remove this. If this is absolutely necessary, we should add a todo comment and say that we need to revisit it in the future